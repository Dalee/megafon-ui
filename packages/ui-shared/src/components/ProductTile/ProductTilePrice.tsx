import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate } from '@megafon/ui-core';
import './style/ProductTilePrice.less';
import AnimationValue from './ProductTileValue';

export interface IProductTilePriceProps {
    /** Заголовок */
    title?: string;
    /** Значение */
    value: string;
    /** Дополнительный юнит */
    unitExtra: string;
    /** Значение юнита */
    unitValue: string;
    /** Скидка */
    discount?: string;
}

const cn = cnCreate('mfui-beta-product-tile-price');
class ProductTilePrice extends React.Component<IProductTilePriceProps> {
    static propTypes = {
        title: PropTypes.string,
        value: PropTypes.string.isRequired,
        unitExtra: PropTypes.string,
        unitValue: PropTypes.string,
        discount: PropTypes.string,
    };

    render() {
        const { title, value, unitExtra, unitValue, discount } = this.props;
        const valueElem = <AnimationValue value={(discount || value) + ` ${unitValue}`} />;
        const unitElem = <span className={cn('payment-period')}>{unitExtra}</span>;

        return (
            <div className={cn('', { discount: !!discount })}>
                <span className={cn('old-price')}>
                    <span className={cn('old-price-value')}>
                        {discount ? `${value} ${unitValue}` : ' '}
                    </span>
                </span>
                <span className={cn('actual-price')}>
                    {valueElem} {unitElem}
                </span>
                <div className={cn('discount-condition')}>{title}</div>
            </div>
        );
    }
}

export default ProductTilePrice;
