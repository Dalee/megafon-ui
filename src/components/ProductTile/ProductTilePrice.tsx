import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate } from '../../utils/cn';
import './style/ProductTilePrice.less';
import AnimationValue from './ProductTileValue';

interface IProductTilePriceProps {
    /** value */
    value: string;
    /** unitExtra */
    unitExtra: string;
    /** unitValue */
    unitValue: string;
    /** discount */
    discount?: string;
}

const cn = cnCreate('mfui-product-tile-price');
class ProductTilePrice extends React.Component<IProductTilePriceProps> {
    static propTypes = {
        value: PropTypes.string.isRequired,
        unitExtra: PropTypes.string,
        unitValue: PropTypes.string,
        discount: PropTypes.string,
    };

    render() {
        const { value, unitExtra, unitValue, discount } = this.props;
        const valueElem = <AnimationValue value={(discount || value) + ` ${unitValue}`} />;
        const unitElem = <span className={cn('payment-period')}>{unitExtra}</span>;

        return (
            <div className={cn('', { discount: !!discount })}>
                <div className={cn('discount-condition')}>При покупке новой SIM–карты</div>
                <span className={cn('old-price')}>
                    <span className={cn('old-price-value')}>
                        {discount ? `${value} ${unitValue}` : ' '}
                    </span>
                </span>
                <span className={cn('actual-price')}>
                    {valueElem} {unitElem}
                </span>
            </div>
        );
    }
}

export default ProductTilePrice;
