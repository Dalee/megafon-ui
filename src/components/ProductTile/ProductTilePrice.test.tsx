import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTilePrice from './ProductTilePrice';

const props = {
    value: '23423',
    unitExtra: 'uniteExtra',
    unitValue: 'uniteValue',
    discount: '23423',
};

describe('<ProductTilePrice />', () => {
    it('it renders ProductTilePrice', () => {
        const wrapper = shallow(<ProductTilePrice {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
