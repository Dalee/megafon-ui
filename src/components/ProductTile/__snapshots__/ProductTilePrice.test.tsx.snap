// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTilePrice /> it renders ProductTilePrice 1`] = `
<div
  className="mfui-product-tile-price"
>
  <div
    className="mfui-product-tile-price__discount-condition"
  >
    При покупке новой SIM–карты
  </div>
  <span
    className="mfui-product-tile-price__old-price"
  >
    <span
      className="mfui-product-tile-price__old-price-value"
    >
      23423
       
      uniteValue
    </span>
  </span>
  <span
    className="mfui-product-tile-price__actual-price"
  >
    23423
     
    uniteValue
     
    <span
      className="mfui-product-tile-price__payment-period"
    >
      uniteExtra
    </span>
  </span>
</div>
`;
